observation_portal/common/downtimedb.py

```diff
@@ -34,35 +34,39 @@ def _get_downtime_data():
         return r.json()['results']
 
     @staticmethod
-    def _order_downtime_by_resource(raw_downtime_intervals):
-        ''' Puts the raw downtime interval sets into a dictionary by resource
+    def _order_downtime_by_resource_and_instrument_type(raw_downtime_intervals):
+        ''' Puts the raw downtime interval sets into a dictionary by resource and then by instrument_type or "all"
         '''
         downtime_intervals = {}
         for interval in raw_downtime_intervals:
             resource = '.'.join([interval['telescope'], interval['enclosure'], interval['site']])
             if resource not in downtime_intervals:
-                downtime_intervals[resource] = []
+                downtime_intervals[resource] = {}
+            instrument_type = interval['instrument_type'] if interval['instrument_type'] else 'all'
+            if instrument_type not in downtime_intervals[resource]:
+                downtime_intervals[resource][instrument_type] = []
             start = datetime.strptime(interval['start'], DOWNTIME_DATE_FORMAT).replace(tzinfo=timezone.utc)
             end = datetime.strptime(interval['end'], DOWNTIME_DATE_FORMAT).replace(tzinfo=timezone.utc)
-            downtime_intervals[resource].append({'type': 'start', 'time': start})
-            downtime_intervals[resource].append({'type': 'end', 'time': end})
+            downtime_intervals[resource][instrument_type].append({'type': 'start', 'time': start})
+            downtime_intervals[resource][instrument_type].append({'type': 'end', 'time': end})
 
         for resource in downtime_intervals:
-            downtime_intervals[resource] = Intervals(downtime_intervals[resource])
+            for instrument_type, intervals in downtime_intervals[resource].items():
```

Comment:
Why loop through `.items()` like this? `intervals` is never used.

Reply:
oops i should be using intervals on the next line